http3/client.go

```diff
@@ -58,6 +58,9 @@ type client struct {
 	dialer       dialFunc
 	handshakeErr error
 
+	receivedSettings chan struct{} // closed once the server's SETTINGS frame was processed
+	settings         *Settings     // set once receivedSettings is closed
```

Comment:
Any reason for preferring the pointer over the raw struct?
```
settings Settings
```

Reply:
I find it cleaner to start with `nil`, since this field is only set once the SETTINGS frame is received. Keep in mind that the zero value of `Settings` is valid as well.